ports/stm32/mboot/Makefile

```diff
@@ -9,6 +9,10 @@ BOARD ?= PYBV10
 BOARD_DIR ?= $(abspath ../boards/$(BOARD))
 endif
 
+# Escape spaces in BOARD_DIR
+SPACE = $(subst ,, )
+BOARD_DIR := $(subst $(SPACE),\ ,$(BOARD_DIR))
```

Comment:
Instead of this, is it possible to change line 9 above to `"$(abspath ../boards/$(BOARD))"`?

Reply:
Unfortunately this doesn't work, because file names are appended to the `BOARD_DIR` variable in lines 27 and 38, and the quotes would end up in the middle of the resulting file paths.

Reply:
I feel like it would be better to quote those lines (27 and 38), rather than substitute spaces like this.  At the very least, quoting is consistent with the other patches you have made.

Reply:
The `wildcard` and `include` statements in lines 30 and 38 don't respect quotes. Any space not escaped by a backslash is treated as the beginning of the next argument by these statements. 

While searching around, I was able to find a single-line version of the above fix:

    # Escape spaces in BOARD_DIR (replaces " " by "\ ")
    BOARD_DIR := $(subst $() $(),\ ,$(BOARD_DIR))

However, that version is probably also less readable than the version currently in this PR.

Leaving the `stm32/mboot` makefile incompatible with paths containing spaces in favor of code readability / maintainability may also be an option.

What are your thoughts on this?

Reply:
My main concern here is moving forward: when changes are made to the makefiles, how does one know what needs to be quoted and what needs to have spaces escaped?  Is there a consistent rule that we can apply to know how to properly deal with spaces in filenames?

Related to this, we do not have any CI that tests for spaces in path names, so that's why it's not well supported by the build system.

Note that there has been fixes in the past for spaces in filenames, eg d53a6d58b04e524e4ec5ee937a15cc3932f8ee2f, da71f55e23a97102d241481e410fdb22d7c55758, #3116.